### PR TITLE
Add Cypress workflow

### PR DESCRIPTION
**Pull request by peterp**
We'll now run our Cypress e2e tests for each pull-request! 🥇 

- [ ] Make sure that running e2e tests works well for users.

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Install framework modules
+        run: yarn install --frozen-lockfile
+
+      - name: Create a RedwoodJS app
+        # NOTE: This path could be a problem if GitHub actions change something.
+        run: mkdir ./my-redwood-app && ./tasks/test-tutorial /home/runner/work/redwood/redwood/my-redwood-app
```
Comment (Tobbe) [inline .github/workflows/e2e.yaml:23]: You could make this more resilient by storing the path in a output variable and read it later when needed

Perhaps something like 
```
id: createpath
run: |
  project_path=$(mktemp -d -t redwood)
  ./task/test-tutorial $project_path
  echo "::set-output name=project_path::$project_path"
```

Then, when you need to reference it later you can do
```
working-directory: ${{ steps.createpath.outputs.project_path }}
```
or
```
env:
  CYPRESS_RW_PATH: "${{ steps.createpath.outputs.project_path }}"
```

File: tasks/test-tutorial [hunk 2]
```diff
@@ -16,34 +16,46 @@ yarn install
...
+  # make the e2e tests use the packages from this Redwood Framework Repo.
   ln -s "$SCRIPTPATH/../packages"
 fi
 
 yarn install
-yarn rw dev --fwd="--open=false" & cd "$SCRIPTPATH/e2e"; yarn cypress open --env RW_PATH=$TMP_DIR
+
+if [ -z "$CI" ]
+  then
+  # TODO: Figure out how to run these automatically without opening the cypress runner.
```
Comment (Tobbe) [inline tasks/test-tutorial:57]: You can do it using the cypress npm module, but you're going to have a bad time on Windows. I created this issue almost a year ago... https://github.com/cypress-io/cypress/issues/7367

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (peterp) [reply .github/workflows/e2e.yaml:23]: Oooh nice!

File: tasks/test-tutorial [hunk 3]
```diff
@@ -16,34 +16,46 @@ yarn install
...
-  # build all the packages
+  # build all the packages, but not the TypeScript
+  # since it is slow and we do not need it.
   yarn build:clean && yarn lerna run build:js
 fi
 
 cd $TMP_DIR
 
-if [ -z "$1" ]
+if [[ -z "$1" || "$CREATE_RWJS_PROJECT" == "1" ]];
```
Comment (dac09) [inline tasks/test-tutorial:46]: We still want to link when its not creating a new project right? Or was this intentional. No need to change this, just checking the logic for porting.

File: tasks/test-tutorial (hunk 2 above)
Comment (peterp) [reply tasks/test-tutorial:57]: Cool, I actually realized that the runner is the best experience on my local computer. This script will die, but I'll remove this comment.

File: tasks/test-tutorial (hunk 3 above)
Comment (peterp) [reply tasks/test-tutorial:46]: This is a poor choice for a variable name.

Basically I want more flags to remix what happens here.

In e2e I want to:
- Create a new project in a specific folder.
-  I want to  build and link the framework files.

Locally I want to be able to:
- Create a new project in a specific folder.
- Do not build, link, and install the framework (leave it be.)
OR:
-  Create a new project in a random folder
-  I want to  build and link the framework files.

**Comment by bhagirath1312** [comment]
Cool scriptpath/ .. /package s

File: tasks/test-tutorial (hunk 3 above)
Comment (peterp) [reply tasks/test-tutorial:46]: Maybe we should just make these flags?

`./run-e2e-tests --new-project --rwt-link`

Comment (dac09) [reply tasks/test-tutorial:46]: Wanna make suggestions on my PR here? https://github.com/redwoodjs/redwood/pull/1896 so I can add them in?

**Comment by peterp** [comment]
I'm not entirely sure why these are timing out now. :/

**Comment by dac09** [comment]
If I had to guess on the timeouts, it could be that the binaries need a chmod. I’ve seen this happen locally

maybe try chmod +x node_modules/.bin/* ? I’ll do it properly in the js version

 (so yarn rw dev can run).

File: .github/workflows/e2e.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+name: Cypress E2E tests
+
+on:
+  push:
+    branches: [main]
```
Comment (thedavidprice) [inline .github/workflows/e2e.yaml:5]: Ideally, for all our CI checks we'd have some logic here:
if (merged via PR) {
    do not trigger
} else {
    assume it's a direct push to main and trigger
}

I think I've wasted at least an hour or two trying to figure out GitHub even logic to make it work.

However, we don't actually have visibility into any checks that run outside of a PR — there are no notifications. Unless we were to set up something like a Slack channel for failure alerts, my vote at this time is to remove this trigger on push to main.

**Comment by thedavidprice** [comment]
Skip my final review approval for this one — I'm good when you gents are good!

I'll make sure I spend time on #1896

**Pull request merged**